onnxruntime/core/providers/cpu/math/gemm_helper.h

```diff
@@ -50,17 +50,17 @@ class GemmHelper {
   Status State() const { return status_; }
 
  private:
-  bool IsValidBroadcast(const TensorShape& shape, int64_t M, int64_t N) {
-    if (shape.NumDimensions() != 1 && shape.NumDimensions() != 2)
+  bool IsValidBroadcast(const TensorShape& bias_shape, int64_t M, int64_t N) {
+    if (bias_shape.NumDimensions() != 0 && bias_shape.NumDimensions() != 1 && bias_shape.NumDimensions() != 2)
```

Comment:
should support scalar bias as well (i.e.) bias_shape.NumDimensions() == 0

Reply:
Could this just be bias_shape.NumDimension() > 2

---
In reply to: [328366840](https://github.com/microsoft/onnxruntime/pull/1923#discussion_r328366840) [](ancestors = 328366840)

Reply:
Sure. Will change this.